packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -167,6 +178,17 @@ export default defineComponent({
         'z-index': props.zIndex && Number(props.zIndex) - 1,
       } as StyleValue
     })
+    const computedBlurStyles = computed(() => {
+      // NOTE Not sure exactly what that does.
+      // Supposedly solves some case when background wasn't shown.
+      // As a side effect removes background from nested modals.
+
+      if (!props.blur) { return }
+
+      return {
+        'backdrop-filter': `blur(${props.blurOpacity}rem)`,
```

Comment:
Backdrop-filter doesn't work in Firefox. The original idea was to use filter on modal's target.

This way blur will not multiply on nested modals.